controllers/nstemplatetier/nstemplatetier_controller.go

```diff
@@ -221,3 +231,38 @@ func NewTTR(tierTmpl *toolchainv1alpha1.TierTemplate, nsTmplTier *toolchainv1alp
 
 	return ttr
 }
+
+func (r *Reconciler) setStatusFailed(ctx context.Context, tmplTier *toolchainv1alpha1.NSTemplateTier, reason string, cause error) error {
+	if err := r.updateStatus(
+		ctx,
+		tmplTier,
+		toolchainv1alpha1.Condition{
```

Comment:
Minor, if we can move this condition out as a separate func or variable it will simplify the code, similar to `ReadyCondition()` you have done

Reply:
makes sense , done in https://github.com/codeready-toolchain/host-operator/pull/1149/commits/45404cdf31313c45a6c6437aaf90bf4cc2199775